search-commons/src/main/java/no/unit/nva/search2/OpenSearchAwsClient.java

```diff
@@ -1,124 +1,142 @@
-import org.apache.http.HttpHost;
-import org.opensearch.action.search.SearchRequest;
-import org.opensearch.action.search.SearchResponse;
-import org.opensearch.client.RequestOptions;
-import org.opensearch.client.RestClient;
-import org.opensearch.client.RestHighLevelClient;
+import org.apache.http.HttpStatus;
 import org.opensearch.common.collect.Tuple;
 import org.opensearch.index.query.Operator;
 import org.opensearch.index.query.QueryBuilders;
-import org.opensearch.index.query.QueryStringQueryBuilder;
 import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.search.sort.SortOrder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-@SuppressWarnings("PMD.CloseResource")
-public class OpenSearchAwsClient implements OpenSearchClient<SearchResponse, ResourceAwsQuery>, Closeable {
+public class OpenSearchAwsClient implements OpenSearchClient<OpenSearchSwsResponse, ResourceAwsQuery> {
 
-    private static final HttpHost httpHost = HttpHost.create(SEARCH_INFRASTRUCTURE_API_URI);
     private static final Logger logger = LoggerFactory.getLogger(OpenSearchAwsClient.class);
     private final CachedJwtProvider jwtProvider;
-    private final RestHighLevelClient client;
+    private final HttpClient httpClient;
+    private final BodyHandler<String> bodyHandler;
 
-    public OpenSearchAwsClient(CachedJwtProvider cachedJwtProvider, RestHighLevelClient client) {
+    public OpenSearchAwsClient(CachedJwtProvider cachedJwtProvider, HttpClient client) {
         super();
         this.jwtProvider = cachedJwtProvider;
-        this.client = client;
+        this.httpClient = client;
+        this.bodyHandler = HttpResponse.BodyHandlers.ofString(StandardCharsets.UTF_8);
     }
 
     @JacocoGenerated
     public static OpenSearchAwsClient defaultClient() {
         var cachedJwtProvider =
             OpenSearchClient
                 .getCachedJwtProvider(new SecretsReader());
-        var client = new RestHighLevelClient(
-            RestClient.builder(httpHost)
-        );
+        var client = HttpClient.newHttpClient();
 
         return new OpenSearchAwsClient(cachedJwtProvider, client);
     }
 
     @Override
-    public SearchResponse doSearch(ResourceAwsQuery query, String mediaType) {
+    public OpenSearchSwsResponse doSearch(ResourceAwsQuery query, String mediaType) {
         return
-            getQueryBuilderStream(query)
+            getQueryBuilderStream(query, mediaType)
                 .map(this::searchSourceWithAggregation)
-                .map(searchSource -> searchSource
-                                         .size(query.getValue(SIZE).as())
-                                         .from(query.getValue(FROM).as())
-                                         .sort(query.getValue(SORT).<String>as()))
-                .map(this::searchRequest)
-                .map(this::searchResponse)
+                .map(this::httpRequest)
+                .map(this::sendHttpRequest)
```

Comment:
maybe just do a `map(this::handleResponse)` here instead of implicit in the request method, makes it more readable, but not necessary